### ISSUE TITLE
preference key used in getter different from setter

### ISSUE THREAD

**Issue by ffleandro** [RaunySouza/preferencer#4]
The following code:
```
@SharedPreference
public abstract class AbstractSessionStore {

    public void setUser(String name, String email, String profilePicture, String authToken, String rememberToken) {
        setName(name);
        setEmail(email);
        setProfilePicture(profilePicture);
        setAuthToken(authToken);
        setRememberToken(rememberToken);
    }

    @Preference
    public abstract String getName();

    @Preference
    public abstract String getProfilePicture();

    @Preference
    public abstract String getEmail();

    @Preference
    public abstract String getAuthToken();

    @Preference
    public abstract String getRememberToken();

    public abstract void setName(String name);
    public abstract void setProfilePicture(String profilePicture);
    public abstract void setEmail(String email);
    public abstract void setAuthToken(String authToken);
    public abstract void setRememberToken(String rememberToken);

}
```

Produces the following `getter` and `setter`:
```
@Override
    public String getAuthToken() {
        return this.preferences.getString("authToken", "");
    }

    public void setAuthToken(String authToken) {
        SharedPreferences.Editor editor = currentTransaction != null ? currentTransaction.editor : preferences.edit();
        editor.putString("auth_token", authToken);
        if (currentTransaction == null) {
            editor.apply();
        }
    }
```

In the getter a `camel cased` key is used but in the setter an `_` separated word key is used.

**Comment by RaunySouza** [comment]
It's strange. They use the same property. I'll check it out.

Other thing, I'm thinking to change the API, mainly the generated code.
For example:

``` java
@SharedPreference
public interface SessionStore {
    @DefaultString("defaultValue")
    String name();
}
```

It'll generate:
``` java
public final class SessionStorePreference extends BaseSharedPreference {
    public StringPreference name() {
        return createStringPreference("name", "defaultValue");
    }

    // And many others boilerplate code.
}
```

This way you'll lost the ability to call other methods inside the base class (Only interfaces will be supported). In the other hand, the amount of generated code will be decreased, and the code will be more organized.

To change the property value:

```java
SessionStorePreference pref = SessionStorePreference.getInstance(context);
pref.name().putIfAbsent("NewValue");
//or
pref.name().remove();
//or
pref.name().get();
```

Instead:

``` java
pref.putName("NewValue");
//or
pref.removeName();
//or
pref.getName();
```

**Comment by ffleandro** [comment]
The docs indicate to use the tag `0.0.1-alpha2` but this issue is fixed in `0.0.1-alpha3`.
